admin-dev/themes/new-theme/js/components/modal.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * ConfirmModal component
+ *
+ * @param {String} id
+ * @param {String} title
+ * @param {String} message
+ * @param {String} closeButtonText
+ * @param {String} confirmButtonText
+ * @param {String} confirmButtonClass
+ * @param {Boolean} closable
+ * @param {Function} action
+ *
+ */
+export default function ConfirmModal({id = 'confirm_modal', title = '', message = '', closeButtonText = 'Close', confirmButtonText = 'Accept', confirmButtonClass = 'btn-primary', closable = true}, action) {
+  // Main modal element
+  this.modal = document.createElement('div');
+  this.modal.classList.add('modal', 'fade');
+  this.modal.id = id;
+
+  // Modal dialog element
+  this.modalDialog = document.createElement('div');
+  this.modalDialog.classList.add('modal-dialog');
+
+  // Modal content element
+  this.modalContent = document.createElement('div');
+  this.modalContent.classList.add('modal-content');
+
+  // Modal header element
```

Comment:
Only one feature missing compared to before, if the title is empty the header should be hidden (display: none) or not constructed (I used a toggle based on title.length)
Be cause an empty header is too big and not pretty